idunn/blocks/reviews.py

```diff
@@ -1,43 +1,82 @@
-from typing import Literal, List
-from pydantic import BaseModel
+from datetime import datetime
+from typing import Literal, List, Optional
+from pydantic import BaseModel, validator
+from idunn.utils.thumbr import thumbr
 
 from .base import BaseBlock
 
+MAX_REVIEW_DISPLAY_NUMBER = 3
+
 
 class Review(BaseModel):
     date: str
+    rating: str
+    # rating_bubble_star_url is not store is ES but build for UI with pydantic validator
+    rating_bubble_star_url: str
     url: str
+    more_reviews_url: str
     lang: str
     title: str
     text: str
-    trip_type: str
+    trip_type: Optional[str]
     author_name: str
 
+    @validator("rating_bubble_star_url", pre=True, always=True)
+    def build_rating_bubble_star_url(cls, rating):
+        # Tripadvisor bubble star url need a rating with exactly one decimal point
+        # (e.g 4.0 or 4.5)
+        rating = f"{float(rating):.1f}"
+
+        base_url = (
+            r"https://www.tripadvisor.com/img/cdsi/img2/ratings/traveler/"
+            f"s{rating}-MCID-66562.svg"
+        )
+
+        if thumbr.is_enabled():
+            return thumbr.get_url_remote_thumbnail(base_url)
+
+        return base_url
+
+
+def build_review(review: dict, source_url: str) -> Review:
+    return Review(
+        date=review["DatePublished"],
+        rating=review["Rating"],
+        rating_bubble_star_url=review["Rating"],
+        url="".join([source_url, review["ReviewURL"]]),
+        more_reviews_url="".join([source_url, review["MoreReviewsURL"]]),
+        lang=review["Language"],
+        title=review["Title"],
+        text=review["Text"],
+        trip_type=review["TripType"],
+        author_name=review["Author"]["AuthorName"],
+    )
+
 
 class ReviewsBlock(BaseBlock):
     type: Literal["reviews"] = "reviews"
     reviews: List[Review]
 
     @classmethod
-    def build_reviews(cls, place):
-        reviews = []
-        for review_dict in place.get_reviews():
-            reviews.append(
-                Review(
-                    date=review_dict["DatePublished"],
-                    url="".join([place.get_source_url(), review_dict["ReviewURL"]]),
-                    lang=review_dict["Language"],
-                    title=review_dict["Title"],
-                    text=review_dict["Text"],
-                    trip_type=review_dict["TripType"],
-                    author_name=review_dict["Author"]["AuthorName"],
-                )
-            )
-        return reviews
+    def build_reviews(cls, reviews: List[dict], source_url: str, lang: str) -> List[Review]:
+        lang_priority_order = {lang: 0, "en": 1}
+        sorted_reviews = sorted(
+            sorted(
+                reviews,
+                key=lambda x: datetime.strptime(x["DatePublished"][:-5], "%Y-%m-%dT%H:%M:%S.%f"),
+                reverse=True,
+            ),
+            key=lambda x: lang_priority_order.get(x["Language"], 2),
+        )
```

Comment:
```suggestion
        sorted_reviews = sorted(
            reviews,
            reverse=True,
            key=lambda x: (
                -lang_priority_order.get(x["Language"], 2),
                datetime.strptime(x["DatePublished"][:-5], "%Y-%m-%dT%H:%M:%S.%f"),
            ),
        )
```

This feels more readable but I'm not sure :see_no_evil: